generic-examples/databases/H2DBMultiDatasources.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+// camel-k: language=java
+// camel-k: dependency=mvn:io.quarkus:quarkus-jdbc-h2
+// camel-k: build-property=quarkus.datasource.default.db-kind=h2
+// camel-k: property=quarkus.datasource.default.username=username-default
+// camel-k: property=quarkus.datasource.default.jdbc.url=jdbc:h2:mem:default
+// camel-k: build-property=quarkus.datasource.testing.db-kind=h2
+// camel-k: property=quarkus.datasource.testing.username=username-testing
+// camel-k: property=quarkus.datasource.testing.jdbc.url=jdbc:h2:mem:testing
+// camel-k: trait=camel.runtime-version=3.2.0
+
+import io.agroal.api.AgroalDataSource;
+import io.quarkus.agroal.DataSource;
+import org.apache.camel.builder.RouteBuilder;
+import jakarta.inject.Inject;
+
+public class H2DBMultiDatasources extends RouteBuilder {
+
+    @Inject
+    @DataSource("default")
+    AgroalDataSource dataSource1;
```

Comment:
I was curious about having to inject the `DataSource`. Is it definitely required?

I tested with a plain Camel Quarkus app earlier today and it was enough to just refer to the DS by name via the `dataSource` URI param.

Reply:
Yeah, I agree that there is no need to delcare the dataSource here.